packages/core/src/configuration/Configuration.spec.ts

```diff
@@ -0,0 +1,146 @@
+        credentials: {
+          token: 'xxxxxxx.xxxx.xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx'
+        }
+      });
+      const configuration2 = configuration.container.resolve(Configuration);
+      expect(configuration).toBe(configuration2);
+    });
+
+    it('should throw if cluster is not passed', () => {
+      expect(
+        () =>
+          new Configuration({
+            cluster: '',
+            credentials: {
+              token: 'xxxxxxx.xxxx.xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx'
+            }
+          })
+      ).toThrow();
+    });
+
+    it('should throw an error if credentials or credential providers are not passed', () => {
+      expect(
+        () =>
+          new Configuration({
+            cluster: 'example.com'
+          })
+      ).toThrow();
+    });
+
+    it.each([
+      {
+        input: 'localhost',
+        expected: { bus: 'amqp://localhost:5672', api: 'http://localhost:8000' }
+      },
+      {
+        input: 'localhost:8080',
+        expected: { bus: 'amqp://localhost:5672', api: 'http://localhost:8000' }
+      },
+      {
+        input: 'http://localhost',
+        expected: { bus: 'amqp://localhost:5672', api: 'http://localhost:8000' }
+      },
+      {
+        input: 'http://localhost:8080',
+        expected: { bus: 'amqp://localhost:5672', api: 'http://localhost:8000' }
+      },
+      {
+        input: '127.0.0.1',
+        expected: { bus: 'amqp://127.0.0.1:5672', api: 'http://127.0.0.1:8000' }
+      },
+      {
+        input: '127.0.0.1:8080',
+        expected: { bus: 'amqp://127.0.0.1:5672', api: 'http://127.0.0.1:8000' }
+      },
+      {
+        input: 'http://127.0.0.1',
+        expected: { bus: 'amqp://127.0.0.1:5672', api: 'http://127.0.0.1:8000' }
+      },
+      {
+        input: 'http://127.0.0.1:8080',
+        expected: { bus: 'amqp://127.0.0.1:5672', api: 'http://127.0.0.1:8000' }
+      },
+      {
+        input: 'example.com',
+        expected: {
+          bus: 'amqps://amq.example.com:5672',
+          api: 'https://example.com'
+        }
+      },
+      {
+        input: 'example.com:443',
+        expected: {
+          bus: 'amqps://amq.example.com:5672',
+          api: 'https://example.com'
+        }
+      },
+      {
+        input: 'http://example.com',
+        expected: {
+          bus: 'amqps://amq.example.com:5672',
+          api: 'https://example.com'
+        }
+      },
+      {
+        input: 'http://example.com:443',
+        expected: {
+          bus: 'amqps://amq.example.com:5672',
+          api: 'https://example.com'
+        }
+      }
+    ])('should generate correct api and bus for $input', entry => {
+      const configuration = new Configuration({
+        cluster: entry.input,
+        credentials: {
+          token: 'xxxxxxx.xxxx.xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx'
+        }
+      });
+
+      expect(configuration.bus).toEqual(entry.expected.bus);
+      expect(configuration.api).toEqual(entry.expected.api);
```

Comment:
Assert the entire result, not on every field. It seems we have already discussed that)
```suggestion
      expect(configuration).toMatchObject(entry.expected);
```